cmd/test_load.go

```diff
@@ -0,0 +1,199 @@
+package cmd
+
+import (
+	"archive/tar"
+	"bytes"
+	"fmt"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"go.k6.io/k6/errext"
+	"go.k6.io/k6/errext/exitcodes"
+	"go.k6.io/k6/js"
+	"go.k6.io/k6/lib"
+	"go.k6.io/k6/lib/metrics"
+	"go.k6.io/k6/loader"
+)
+
+const (
+	testTypeJS      = "js"
+	testTypeArchive = "archive"
+)
+
+type loadedTest struct {
+	testPath        string // contains the raw string the user supplied
+	source          *loader.SourceData
+	fileSystems     map[string]afero.Fs
+	runtimeOptions  lib.RuntimeOptions
+	metricsRegistry *metrics.Registry
+	builtInMetrics  *metrics.BuiltinMetrics
+	initRunner      lib.Runner // TODO: rename to something more appropriate
+
+	// Only set if cliConfigGetter is supplied to loadTest() or if
+	// consolidateDeriveAndValidateConfig() is manually called.
+	consolidatedConfig Config
+	derivedConfig      Config
+}
+
+func loadTest(
+	gs *globalState, cmd *cobra.Command, args []string,
+	// supply this if you want the test config consolidated and validated
+	cliConfigGetter func(flags *pflag.FlagSet) (Config, error), // TODO: obviate
+) (*loadedTest, error) {
+	if len(args) < 1 {
+		return nil, fmt.Errorf("k6 needs at least one argument to load the test")
+	}
+
+	testPath := args[0]
```

Comment:
In the spirit of my comment about naming, I would argue that this could be called `sourcePath`, as it's only used for logging and in the context of `readSource`.

Reply:
`rootSourcePath` or `sourceRootPath` will probably be even better, considering a script can `import` other files :thinking: